Marlin/src/lcd/dogm/status_screen_DOGM.cpp

```diff
@@ -413,9 +434,19 @@ void MarlinUI::draw_status_screen() {
         #if ENABLED(SHOW_REMAINING_TIME)
           if (!(ev & 0x3)) {
             duration_t estimation = elapsed.value * (100 * (PROGRESS_SCALE) - progress) / progress;
-            const bool has_days = (estimation.value >= 60*60*24L);
-            const uint8_t len = estimation.toDigital(estimation_string, has_days);
-            estimation_x_pos = _SD_DURATION_X(len + 1);
+            if (estimation.value == 0) {
+              estimation_string[0] = '\0';
+              estimation_x_pos = _SD_DURATION_X(0);
+            }
+            else {
+              const bool has_days = (estimation.value >= 60*60*24L);
```

Comment:
Note that the "`has_days`" (`with_days`) parameter is included only for the sake of using less horizontal space. So the condition here should not be "is this more than 24 hours?" but "do we have enough space to display days, or only enough space to show 25+ hours?".